### ISSUE TITLE
 failed to Optimizing a build for size 

### ISSUE THREAD

**Issue by nagasawamaki** [godotengine/godot#22018]
**Godot version:**
branch 3.0 (should be 3.0.6 now) 

**OS/device including version:**
Windows 7

**Issue description:**
I download the source code from branch 3.0 (should be 3.0.6 now) and run the following:
`scons p=android target=release_debug tools=no disable_3d=yes`

but I get

```
scons: *** [bin\libgodot.android.opt.debug.armv7.neon.so] Error 1
scons: building terminated because of errors.
```

Is the Optimizing a build for size only work for 3.1?
Or am I missing something?
Thank you.

ps. the output of compilation:

```
modules\libmodules.android.opt.debug.armv7.neon.a(soft_body_bullet.android.opt.debug.armv7.neon.o):modules\bullet\soft_body_bullet.cpp:function SoftBodyBullet::SoftBodyBullet(): error: undefined reference to 'ImmediateGeometry::ImmediateGeometry()'

modules\libmodules.android.opt.debug.armv7.neon.a(soft_body_bullet.android.opt.debug.armv7.neon.o):modules\bullet\soft_body_bullet.cpp:function SoftBodyBullet::dispatch_callbacks(): error: undefined reference to 'ImmediateGeometry::clear()'

modules\libmodules.android.opt.debug.armv7.neon.a(soft_body_bullet.android.opt.debug.armv7.neon.o):modules\bullet\soft_body_bullet.cpp:function SoftBodyBullet::dispatch_callbacks(): error: undefined reference to 'ImmediateGeometry::begin(Mesh::PrimitiveType, Ref<Texture> const&)'
modules\libmodules.android.opt.debug.armv7.neon.a(soft_body_bullet.android.opt.debug.armv7.neon.o):modules\bullet\soft_body_bullet.cpp:function SoftBodyBullet::dispatch_callbacks(): error: undefined reference to 'ImmediateGeometry::add_vertex(Vector3 const&)'
modules\libmodules.android.opt.debug.armv7.neon.a(soft_body_bullet.android.opt.debug.armv7.neon.o):modules\bullet\soft_body_bullet.cpp:function SoftBodyBullet::dispatch_callbacks(): error: undefined reference to 'ImmediateGeometry::add_vertex(Vector3 const&)'
modules\libmodules.android.opt.debug.armv7.neon.a(soft_body_bullet.android.opt.debug.armv7.neon.o):modules\bullet\soft_body_bullet.cpp:function SoftBodyBullet::dispatch_callbacks(): error: undefined reference to 'ImmediateGeometry::add_vertex(Vector3 const&)'
modules\libmodules.android.opt.debug.armv7.neon.a(soft_body_bullet.android.opt.debug.armv7.neon.o):modules\bullet\soft_body_bullet.cpp:function SoftBodyBullet::dispatch_callbacks(): error: undefined reference to 'ImmediateGeometry::end()'clang++.exe: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by Calinou** [comment]
As far as I know, the `disable_3d` flag was fixed in the `master` branch, but not in the 3.0 branch.

**Comment by akien-mga** [comment]
Fixed in 3.1.
